extension/source/scriptutil.js

```diff
@@ -0,0 +1,45 @@
+
+const ATTR_ORIGINAL = 'osext-orig';
+
+// This observer is used to store original table cell values (excluding line breaks) in an attribute.
+// If the content has been manipulated by a user script, the page processing can use the original value.
+new MutationObserver((records, observer) => {
+	records.forEach(record => {
+		if (record.target.nodeName == 'TR') {
+			record.addedNodes.forEach(node => {
+				if (node.nodeName == 'TD' && node.textContent.length && !node.textContent.match(/[\r\n]+/g)) {
+					if (!node.getAttribute(ATTR_ORIGINAL)) node.setAttribute(ATTR_ORIGINAL, node.textContent);
+				}
+			});
+		}
+		else if (record.target.nodeName == 'BODY' && record.addedNodes.length > 0 && record.addedNodes.item(0).getAttribute 
+			&& record.addedNodes.item(0).getAttribute('id') == ID_MARKER) {
+			observer.disconnect();
+		}
+	});
+}).observe(document, { childList: true, subtree: true });
+
+/**
+ * Utilities for user script handling.
+ */
+class ScriptUtil {
+
+	constructor () {
+		// utility
+	}
+
+	/**
+	 * Returns the original (or initial) value for the given cell.
+	 *
+	 * @param {HTMLTableCellElement} cell
+	 * @returns {String} value
+	 */
+	static getOriginalCellContent(cell) {
+		if (cell instanceof HTMLTableCellElement) {
+			return cell.getAttribute(ATTR_ORIGINAL) || cell.textContent;
```

Comment:
Dies funktioniert natürlich nur, wenn die Zelle nicht legal leer (`''`, `0`, `null`, etc.) sein kann, z.B. wenn immer Zahlen drin stehen, was hier der Fall ist... Vielleicht ist das mal so eine Stelle, wo man auf `undefined` oder `typeof ... == 'undefined'` überprüfen sollte...

Reply:
Der textContent einer Zelle kann nur String sein, und das Original-Attribut (ebenfalls String) wird nur definiert, wenn es zumindest ein Zeichen gibt.

Reply:
Mir macht der Fall `""` [String] Sorge, insbesondere in der Spalte des Spielplans, in der normalerweise nichts steht, aber das Script die Statistik reinschießt!
Dann wäre doch `cell.getAttribute(ATTR_ORIGINAL) === ""` und somit würde immer noch die Statistik in `cell.textContent` zurückgegeben. Korrekt ist das nicht wirklich (und eine Stolperfalle)...